### PR TITLE
Minor code cleanup for options

### PR DESCRIPTION
**Pull request by Youssef1313**
@tmat @mavasani for review.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public static SyntaxFormattingOptions.CommonOptions GetCommonSyntaxFormattingOpt
         {
             LineFormatting = options.GetLineFormattingOptions(fallbackOptions.LineFormatting),
             SeparateImportDirectiveGroups = options.GetEditorConfigOption(GenerationOptions.SeparateImportDirectiveGroups, fallbackOptions.SeparateImportDirectiveGroups),
-            AccessibilityModifiersRequired = options.GetEditorConfigOptionValue(CodeStyleOptions2.AccessibilityModifiersRequired, fallbackOptions.AccessibilityModifiersRequired),
+            AccessibilityModifiersRequired = options.GetEditorConfigOption(CodeStyleOptions2.AccessibilityModifiersRequired, fallbackOptions.AccessibilityModifiersRequired),
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs:65]: The line just above this one is already calling `GetEditorConfigOption`.
I'm assuming we cannot observe a difference between `GetEditorConfigOption` and `GetEditorConfigOptionValue`, so I removed `GetEditorConfigOptionValue` and called `GetEditorConfigOption` instead.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalyzerConfigOptionsExtensions.cs [hunk 2]
```diff
@@ -55,7 +39,7 @@ private static T GetOptionWithAssertOnFailure<T>(AnalyzerConfigOptions analyzerC
             return value;
         }
 
-        public static bool TryGetEditorConfigOptionOrDefault<T>(this AnalyzerConfigOptions analyzerConfigOptions, TOption option, out T value)
+        private static bool TryGetEditorConfigOptionOrDefault<T>(this AnalyzerConfigOptions analyzerConfigOptions, TOption option, out T value)
             => TryGetEditorConfigOption(analyzerConfigOptions, option, (T?)option.DefaultValue, out value!);
 
         public static bool TryGetEditorConfigOption<T>(this AnalyzerConfigOptions analyzerConfigOptions, TOption option, [MaybeNullWhen(false)] out T value)
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalyzerConfigOptionsExtensions.cs:45]: @mavasani @tmat I'm confused by the implementation of this method. It's passing `default` literal to `defaultValue`. Why we're not always using `option.DefaultValue` (ie, `TryGetEditorConfigOptionOrDefault`)

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs:65]: This change is incorrect. I'll revert.

Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs:65]:
```suggestion
            AccessibilityModifiersRequired = options.GetEditorConfigOptionValue(CodeStyleOptions2.AccessibilityModifiersRequired, fallbackOptions.AccessibilityModifiersRequired),
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeActions/CodeFixOptionsProvider.cs [hunk 3]
```diff
@@ -56,7 +56,7 @@ public LineFormattingOptions GetLineFormattingOptions()
     public SyntaxFormattingOptions GetFormattingOptions(ISyntaxFormatting formatting)
         => formatting.GetFormattingOptions(_options, FallbackSyntaxFormattingOptions);
 
-    public AccessibilityModifiersRequired AccessibilityModifiersRequired => _options.GetEditorConfigOptionValue(CodeStyleOptions2.AccessibilityModifiersRequired, FallbackCommonSyntaxFormattingOptions.AccessibilityModifiersRequired);
+    public AccessibilityModifiersRequired AccessibilityModifiersRequired => _options.GetEditorConfigOption(CodeStyleOptions2.AccessibilityModifiersRequired, FallbackCommonSyntaxFormattingOptions.AccessibilityModifiersRequired);
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeActions/CodeFixOptionsProvider.cs:59]:
```suggestion
    public AccessibilityModifiersRequired AccessibilityModifiersRequired => _options.GetEditorConfigOptionValue(CodeStyleOptions2.AccessibilityModifiersRequired, FallbackCommonSyntaxFormattingOptions.AccessibilityModifiersRequired);
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalyzerConfigOptionsExtensions.cs [hunk 4]
```diff
@@ -64,9 +48,6 @@ public static bool TryGetEditorConfigOption<T>(this AnalyzerConfigOptions analyz
         public static T GetEditorConfigOption<T>(this AnalyzerConfigOptions analyzerConfigOptions, TOption option, T defaultValue)
             => TryGetEditorConfigOption(analyzerConfigOptions, option, new Optional<T?>(defaultValue), out var value) ? value! : throw ExceptionUtilities.Unreachable;
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/AnalyzerConfigOptionsExtensions.cs:50]:
```suggestion

        public static T GetEditorConfigOptionValue<T>(this AnalyzerConfigOptions analyzerConfigOptions, TOption option, T defaultValue)
            => TryGetEditorConfigOption(analyzerConfigOptions, option, new Optional<CodeStyleOption2<T>?>(new CodeStyleOption2<T>(defaultValue, NotificationOption2.None)), out var style) ? style!.Value : throw ExceptionUtilities.Unreachable;
```

**Review by jmarolf — APPROVED** [review]

**Review by genlu — APPROVED** [review]

**Pull request merged**